subprojects/dependency-management/src/main/java/org/gradle/internal/resource/transfer/DefaultCacheAwareExternalResourceAccessor.java

```diff
@@ -213,10 +209,12 @@ private LocallyAvailableExternalResource copyToCache(final ExternalResourceName
     }
 
     private LocallyAvailableExternalResource moveIntoCache(final ExternalResourceName source, final File destination, final ResourceFileStore fileStore, final ExternalResourceMetaData metaData) {
-        LocallyAvailableResource cachedResource = fileStore.moveIntoCache(destination);
-        File fileInFileStore = cachedResource.getFile();
-        cachedExternalResourceIndex.store(source.toString(), fileInFileStore, metaData);
-        return fileResourceRepository.resource(fileInFileStore, source.getUri(), metaData);
+        return artifactCacheLockingManager.useCache(() -> {
```

Comment:
@melix Given the refactoring you did, that seemed like the right thing to do. Can you confirm?